docs/matchers/index.rst

```diff
@@ -0,0 +1,388 @@
+      self.mock_callable(os, 'remove')\
+          .for_call(timeout=matchers.NotThisFloat(6.9))\
+          .to_return_value(None)
+
+FloatBetween
+^^^^^^^^^^^^
+testslide.matchers.FloatBetween() represents an argument, that matches any Float, that falls in the specified range
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(timeout=matchers.FloatBetween(5.9, 6.9))\
+          .to_return_value(None)
+
+
+FloatGreater
+^^^^^^^^^^^^
+testslide.matchers.FloatGreater() represents an argument, that matches any Float, that is greater than the specified one
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(timeout=matchers.FloatGreater(6.9))\
+          .to_return_value(None)
+
+FloatGreaterOrEquals
+^^^^^^^^^^^^^^^^^^^^
+testslide.matchers.FloatGreaterOrEquals() represents an argument, that matches any Float, that is greater than or equals the specified one
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(timeout=matchers.FloatGreaterOrEquals(6.9))\
+          .to_return_value(None)
+
+FloatLess
+^^^^^^^^^
+testslide.matchers.FloatLess() represents an argument, that is less than the specified Float
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(timeout=matchers.FloatLess(6.9))\
+          .to_return_value(None)
+
+FloatLessOrEquals
+^^^^^^^^^^^^^^^^^
+testslide.matchers.FloatLessOrEquals() represents an argument, that is less than or equals the specified Float
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(timeout=matchers.FloatLessOrEquals(6.9))\
+          .to_return_value(None)
+
+
+Strings
+-------
+
+AnyStr
+^^^^^^
+testslide.matchers.AnyStr() represents an argument, that matches any String
+
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(matchers.AnyStr())\
+          .to_return_value(None)
+
+RegexMatches
+^^^^^^^^^^^^
+testslide.matchers.RegexMatches() represents an argument, that is a String and matches provided regular expression.
+
+
+.. code-block:: python
+
+  import testslide, backup, storage, matchers
+  
+    def testrmjson(self):
+      self.mock_callable(os, 'remove')\
+          .for_call(matchers.RegexMatches(".*\.json"))\
+          .to_return_value(None)
+
+
+Collections
+-----------
+
+NotEmpty
+^^^^^^^^
+testslide.matchers.NotEmpty() represents an argument, that is a collection (eg: list, dict, tuple), and has elements.
```

Comment:
I don't think Python has Ruby like `.empty?` stuff, so perhaps it is worth explaining that this is essentially an alias for ``False``.